ext/spl/spl_directory.stub.php

```diff
@@ -142,46 +142,46 @@ class FilesystemIterator extends DirectoryIterator
 {
     public function __construct(string $directory, int $flags = FilesystemIterator::KEY_AS_PATHNAME | FilesystemIterator::CURRENT_AS_FILEINFO | FilesystemIterator::SKIP_DOTS) {}
 
-    /** @return void */
-    public function rewind() {}
+    /** @tentative-return-type */
+    public function rewind(): void {}
 
     /** @return string */
     public function key() {}
 
-    /** @return string|SplFileInfo|self */
+    /** @return string|SplFileInfo|FilesystemIterator */
```

Comment:
Making this a proper type should also be fine?

Reply:
Yes, but it might be problematic (see my previous comment)